nav2_smac_planner/include/nav2_smac_planner/smac_planner_2d.hpp

```diff
@@ -100,6 +100,16 @@ class SmacPlanner2D : public nav2_core::GlobalPlanner
   bool _downsample_costmap;
   rclcpp_lifecycle::LifecyclePublisher<nav_msgs::msg::Path>::SharedPtr _raw_plan_publisher;
   double _max_planning_time;
+
```

Comment:
Hi,

Certainly appreciate this and helps get one of the algorithms done for run time reconfiguration. Something though is that it would be great if this also enabled runtime reconfiguration of any parameter here that can be reconfigured. Some of the parameters are going to be set on initialization and are therefore not reconfigurable, but some are used at runtime and can therefore be reconfigured.

Reply:
- _downsampling_factor
- _max_planning_time

Seem like the only 2 I can find

Reply:
`_downsampling_factor` is used to initialize the `CostmapDownsampler`
https://github.com/ros-planning/navigation2/blob/5c61644651c4eab882b042e073d0f5964f03a501/nav2_smac_planner/src/smac_planner_2d.cpp#L139-L145

I see only `_max_planning_time` okay to modify at runtime in the same way as `tolerance`.
But we could allow reconfigurability of all parameters if we trigger re-initialization of the relevant objects. But it will require a mechanism ensuring that no planning is currently being computed. What do you think ?

Reply:
Its also used in the call to down sample https://github.com/ros-planning/navigation2/blob/b519dece8b5f1268e8bb16ca9b1b04fa06c75e34/nav2_smac_planner/src/smac_planner_2d.cpp#L215 

Frankly, I don't even remember what I was thinking where I wanted to have that in the configuration argument if its also in the `downsample()` function argument. It might be able to be safely removed.

Reply:
> But we could allow reconfigurability of all parameters if we trigger re-initialization of the relevant objects

Certainly, yes. We could add a mutex lock to prevent planning during these reconfiguration phases.

Reply:
Alright, I will take care of it with mutex lock when I will have a minute.

Reply:
curious -- are you guys using Smac 2D?

Reply:
I started evaluating it (on a real robot/environment) today ;)
Not yet in production though but it is a possibility

Reply:
Let me know if that changes. The Hybrid-A* version is actually faster in many cases -- and it is kinematically feasible so no plans starting at a 60 deg angle from your starting orientation.

Reply:
I thought the 2D version was more appropriate to small circular base ?
Plus at first inspection, there is no tolerance parameter on the Hybrid-A* version.
But I am already a fan of the path cleanness the 2D version produce, despite the reduced efficiency (we compute global path only once in our use case and do recomputation only in case of local planner/controller failure)

Reply:
> I thought the 2D version was more appropriate to small circular base ?

Sure, but its not kinematically feasible, the Hybrid-A* planner is. The "tolerance" isn't as important because if there is a way to get to the goal, the analytic expansions make it much easier to make sure its possible. Use what is better for your situation, but I don't think you should dismiss Hybrid-A* out of hand. No more 'rotate to heading' to follow a path, the path will always start from the robot's heading smoothly.

Reply:
Okay, I will have a look a some point. For our current use case (circular robot), 'rotate to heading' is perfectly acceptable.